src/track/track_gps_l2cm.c

```diff
@@ -0,0 +1,614 @@
+    log_warn_sid(sid, "Unexpected L1C/A to L2C handover code phase: %f",
+                 code_phase);
+    return;
+  }
+
+  if (code_phase > (GPS_L1CA_CHIPS_NUM - 0.5)) {
+    code_phase = 2 * GPS_L2CM_CHIPS_NUM - (GPS_L1CA_CHIPS_NUM - code_phase);
+  }
+
+  /* recalculate doppler freq for L2 from L1*/
+  double carrier_freq = tracking_channel_carrier_freq_get(nap_channel) *
+                        GPS_L2_HZ / GPS_L1_HZ;
+
+  /* get initial cn0 from parent L1 channel */
+  float cn0_init = tracking_channel_cn0_get(nap_channel);
+
+  /* The best elevation estimation could be retrieved by calling
+     tracking_channel_evelation_degrees_get(nap_channel) here.
+     However, we assume it is done where tracker_channel_init()
+     is called. */
+
+  tracking_startup_params_t startup_params = {
+    .sid                = sid,
+    .sample_count       = sample_count,
+    .carrier_freq       = carrier_freq,
+    .code_phase         = code_phase,
+    .chips_to_correlate = L2CM_TRACK_SHORT_CYCLE_INTERVAL_CHIPS,
+    .cn0_init           = cn0_init,
+    .elevation          = TRACKING_ELEVATION_UNKNOWN
+  };
+
+  if (tracking_startup_request(&startup_params)) {
+    log_info_sid(sid, "L2 CM handover done");
+  } else {
+    log_error_sid(sid, "Failed to start L2C tracking");
+  }
+}
+
+static void tracker_gps_l2cm_init(const tracker_channel_info_t *channel_info,
+                                  tracker_common_data_t *common_data,
+                                  tracker_data_t *tracker_data)
+{
+  (void)channel_info;
+  gps_l2cm_tracker_data_t *data = tracker_data;
+
+  memset(data, 0, sizeof(gps_l2cm_tracker_data_t));
+  tracker_ambiguity_unknown(channel_info->context);
+
+  const struct loop_params *l = &loop_params_stage;
+
+  assert(20 == l->coherent_ms);
+  data->int_ms = l->coherent_ms;
+
+  aided_tl_init(&(data->tl_state), 1e3 / data->int_ms,
+                common_data->code_phase_rate - GPS_CA_CHIPPING_RATE,
+                l->code_bw, l->code_zeta, l->code_k,
+                l->carr_to_code,
+                common_data->carrier_freq,
+                l->carr_bw, l->carr_zeta, l->carr_k,
+                l->carr_fll_aid_gain);
+
+  data->short_cycle = true;
+  data->startup = 2;
+
+  /* Initialise C/N0 estimator */
+  cn0_est_init(&data->cn0_est, 1e3 / data->int_ms, common_data->cn0,
+               CN0_EST_LPF_CUTOFF, 1e3 / data->int_ms);
+
+  /* Initialize lock detector */
+  lock_detect_init(&data->lock_detect,
+                   lock_detect_params.k1, lock_detect_params.k2,
+                   lock_detect_params.lp, lock_detect_params.lo);
+
+  /* TODO: Reconfigure alias detection between stages */
+  u8 alias_detect_ms = l->coherent_ms;
+  alias_detect_init(&data->alias_detect,
+                    L2C_ALIAS_DETECT_INTERVAL_MS / alias_detect_ms,
+                    (alias_detect_ms - 1) * 1e-3);
+
+  /* Let's remember SV ID to avoid double tracking of L2C from it. */
+  data->sat = channel_info->sid.sat;
```

Comment:
My bad. Should not be here at all. It is a leftover from the older approach, where I checked if the SV L2CM signal is already in track. Anyways, now it is removed.